vignettes/poisson.Rmd

```diff
@@ -0,0 +1,359 @@
+`logability` between a team and its opponent is a useful predictor for the
+number of `goals` scored.
+
+Consequently, we fit a generalized linear model (GLM) to the data that links
+the expected number of goals per team/match $\lambda_i$ to the linear predictor
+$x_i^\top \beta$ with regressor vector $x_i^\top = (1, \mathtt{difference}_i)$
+and corresponding coefficient vector $\beta$ using a log-link:
+$\log(\lambda_i) = x_i^\top \beta$. The maximum likelihood estimator $\hat \beta$
+with corresponding inference, predictions, residuals, etc. can be obtained in R
+using the `glm()` function with `family = poisson`:
+
+```{r}
+m <- glm(goals ~ difference, data = FIFA2018, family = poisson)
+summary(m)
+```
+
+Both parameters have intuitive interpretations. First, the intercept
+corresponds to the expected log-goals per team in a match of two equally strong teams,
+i.e., with zero difference in log-abilities. The corresponding prediction for the
+number of goals can either be obtained manually from the extracted `coef()`
+or equivalently using the `predict()` function:
+
+```{r}
+lambda_zero <- exp(coef(m)[1])
+lambda_zero
+predict(m, newdata = data.frame(difference = 0), type = "response")
+```
+
+As above, we can also set up a `Poisson()` distribution object and obtain the
+associated expected probability distribution for zero to six goals:
+
+```{r}
+p_zero <- Poisson(lambda = lambda_zero)
+pdf(p_zero, 0:6)
+```
+
+Second, the slope of $`r round(coef(m)[2], digits = 3)`$ can be interpreted
+as an ability elasticity of the number of goals scored. This is because the
+difference of the log-abilities can also be understood as the log of the ability
+ratio. Thus, when the ability ratio increases by $1$ percent, the expected
+number of goals increases approximately by $`r round(coef(m)[2], digits = 3)`$
+percent.
+
+This yields a different predicted Poisson distribution for each team/match
+in the tournament. We can easily set up the vector of all $128$ `Poisson()`
+distribution objects by extracting the vector of all fitted point estimates
+$(\hat \lambda_1, \dots, \hat \lambda_{128})^\top$:
+
+```{r}
+p_reg <- Poisson(lambda = fitted(m))
+head(p_reg)
+```
+
+As an illustration, the following goal distributions could be expected for the
+final that France won 4-2 against Croatia:
+
+```{r}
+tail(FIFA2018, 2)
+p_final <- tail(p_reg, 2)
+p_final
+pdf(p_final, 0:6)
+```
+
+Clearly, France was expected to score more goals than Croatia
+but both teams scored more goals than expected, albeit not unlikely many.
+
+
+## Further details and extensions
+
+Assuming independence of the number of goals scored, we can obtain
+the table of possible match results (after normal time) by multiplying
+the marginal probabilities (again only up to six goals). In R this be done
+easily using the `outer()` function which by default performs a multiplication
+of its arguments.
+
+```{r}
+res <- outer(pdf(p_final[1], 0:6), pdf(p_final[2], 0:6))
```

Comment:
Clever, the fact that goals by each team are approximately independent is fascinating.

Reply:
Indeed it is. One could illustrate this by something like:

```
r <- residuals(m)
cor(r[1:64 * 2 - 1], r[1:64 * 2])
```

but I thought this was maybe too cryptic.